service/src/main/kotlin/fi/espoo/evaka/application/ApplicationStateService.kt

```diff
@@ -96,6 +99,66 @@ class ApplicationStateService(
 ) {
     private val filesBucket = env.attachments
 
+    fun initializeApplicationForm(
+        tx: Database.Transaction,
+        user: AuthenticatedUser,
+        applicationId: ApplicationId,
+        type: ApplicationType,
+        guardian: PersonDTO,
+        child: PersonDTO
+    ) {
+        val form = ApplicationForm.initForm(type, guardian, child)
+            .let { form -> withDefaultStartDate(tx, type, form) }
+            .let { form -> withDefaultOtherChildren(tx, user, guardian, child, form) }
+
+        tx.updateForm(
+            applicationId,
+            form,
+            type,
+            child.restrictedDetailsEnabled,
+            guardian.restrictedDetailsEnabled
+        )
+    }
+
+    private fun withDefaultStartDate(
+        tx: Database.Transaction,
+        type: ApplicationType,
+        form: ApplicationForm
+    ): ApplicationForm {
+        val today = dateNow()
+
+        val startDate = when (type) {
+            ApplicationType.PRESCHOOL -> tx.getPreschoolTerms().firstOrNull { it.extendedTerm.start.isAfter(today) }
+                ?.extendedTerm?.start
+            ApplicationType.CLUB -> tx.getClubTerms().firstOrNull { it.term.start.isAfter(today) }
+                ?.term?.start
```

Comment:
Should one still be able to apply to club after term has started? I don't remember

Reply:
Yeah, but this only affects the default date. So if the term has started we don't fill a default start date.